doc/api-extensions.md

```diff
@@ -2273,3 +2273,6 @@ This allows copying storage volume snapshots to and from remotes.
 ## `zfs_delegate`
 This implements a new `zfs.delegate` volume Boolean for volumes on a ZFS storage driver.
 When enabled and a suitable system is in use (requires ZFS 2.2 or higher), the ZFS dataset will be delegated to the container, allowing for its use through the `zfs` command line tool.
+
+## `generated_documentation`
```

Comment:
Can we come up with a better, more descriptive name for this? How about "documentation_config"?

Also shouldn't we hold off merging this until all of the entities have been documented?

Reply:
Yep, we first need to merge the documented codebase. I agree, we need to change the name.

Reply:
OK thanks. Please can we include all the documentation changes in this PR so that they can be merged at the same time as the API extension. Any precursor tooling or hook changes that dont actually depend on the content or api extension can be in a separate PR.